Samples/Samples/View/CalendarPage.xaml

```diff
@@ -1,49 +1,61 @@
 ﻿<?xml version="1.0" encoding="utf-8" ?>
-<views:BasePage xmlns="http://xamarin.com/schemas/2014/forms"
+<views:BasePage  xmlns="http://xamarin.com/schemas/2014/forms"
              xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
              xmlns:views="clr-namespace:Samples.View"
              xmlns:viewmodels="clr-namespace:Samples.ViewModel"
              xmlns:essentials="clr-namespace:Xamarin.Essentials;assembly=Xamarin.Essentials"
              xmlns:samples="clr-namespace:Samples.ViewModel;assembly=Samples"
              x:Class="Samples.View.CalendarPage"
-             Title="Calendar">
-    <views:BasePage.BindingContext>
-        <viewmodels:CalendarViewModel />
-    </views:BasePage.BindingContext>
-    <views:BasePage.Resources>
-        <ResourceDictionary>
-            <DataTemplate x:Key="CalendarItemTemplate">
-                <ViewCell>
-                    <StackLayout>
-                        <Label Text="{Binding Name}" TextColor="Black" HorizontalTextAlignment="Center"/>
-                    </StackLayout>
-                </ViewCell>
-            </DataTemplate>
-        </ResourceDictionary>
+             Title="Calendars">
+  <views:BasePage.Resources>
+      <ResourceDictionary>
+          <DataTemplate x:Key="EventItemTemplate">
+              <ViewCell>
+                  <StackLayout>
+                      <Label Text="{Binding Title}"
+                             TextColor="Black"
+                             HorizontalTextAlignment="Center"/>
+                  </StackLayout>
+              </ViewCell>
+          </DataTemplate>
+      </ResourceDictionary>
     </views:BasePage.Resources>
-    <Grid>
+  <views:BasePage.BindingContext>
+      <viewmodels:CalendarViewModel />
+    </views:BasePage.BindingContext>
+  <views:BasePage.Content>
+      <Grid>
         <Grid.RowDefinitions>
-            <RowDefinition Height="Auto"/>
-            <RowDefinition/>
+          <RowDefinition Height="Auto"/>
+          <RowDefinition Height="Auto"/>
+          <RowDefinition/>
         </Grid.RowDefinitions>
         <Grid>
-            <Grid.ColumnDefinitions>
-                <ColumnDefinition/>
-                <ColumnDefinition/>
-                <ColumnDefinition/>
-            </Grid.ColumnDefinitions>
-            <Button Text="Request Read Access"
-                    Command="{Binding RequestCalendarReadAccess}"
-                    HeightRequest="100"/>
-            <Button Grid.Row="0" Grid.Column="1" 
-                    Text="Request Write Access"
-                    Command="{Binding RequestCalendarWriteAccess}"/>
-            <Button Grid.Row="0" Grid.Column="2"  
-                    Text="Load Calendars" 
-                    Command="{Binding GetCalendars}"/>
+          <Grid.ColumnDefinitions>
+            <ColumnDefinition/>
+            <ColumnDefinition/>
+            <ColumnDefinition/>
+          </Grid.ColumnDefinitions>
+          <Button Text="Request Read Access"
```

Comment:
When this page loads, we should query to see if we have read permissions. If we do, we should change the UI to do something like hide the button and/or indicate which permissions the user has granted

Reply:
Was having a look into this one. The only issue at the moment is that PermissionStatus is not public so I'm unable to return the enum result from the function:
internal static Task<PermissionStatus> CheckStatusAsync(PermissionType permission) =>
PlatformCheckStatusAsync(permission);

every sample only seems to use:
internal static async Task RequireAsync(PermissionType permission)
        {
            if (await RequestAsync(permission) != PermissionStatus.Granted)
                throw new PermissionException($"{permission} was not granted.");
        }

(as they have no check permission status calls)

Which leaves me with a few options:
A (bool) - make a call similar to the one above but just say to return true if Granted
B (PermissionType) - Allow permission type to be publicly accessible, Microsoft might not agree.
c (local PermissionType) - Replicate the PermissionType enum in my sample allowing me to return it without exposing their private enum.